R/MsBackendTimsTof.R

```diff
@@ -29,18 +38,39 @@ setMethod("backendInitialize", signature = "MsBackendTimsTof",
             msg <- Spectra:::.valid_ms_backend_files_exist(files)
             if (length(msg))
               stop(msg)
-            # despite the code above I assume that files is a character(1) 
-            # because maybe it's better to discuss the multiple files case 
-            # before.
-            object <- .header(object, files) 
-            # validObject(object)
+            object <- .initialize(object, files, BPPARAM) 
+            validObject(object)
             object
           })
 
-
 #' @exportMethod length
 #'
 #' @rdname MsBackendTimsTof
 setMethod("length", "MsBackendTimsTof", function(x) {
-  nrow(x@indexes)
+  nrow(x@indices)
 })
+
+
+#' @importFrom MsCoreUtils i2index
+#'
+#' @rdname hidden_aliases
+setMethod("[", "MsBackendTimsTof", function(x, i, j, ..., drop = FALSE) {
+  if (missing(i))
+    return(x)
+  i <- i2index(i, length(x))
+  slot(x, "indices", check = FALSE) <- x@indices[i, , drop = FALSE]
+  slot(x, "frames", check = FALSE) <- 
+    x@frames[match(unique(paste0(x@indices[, "frame"], x@indices[, "file"])),
+                   paste0(x@frames$Id, x@frames$file)), , drop = FALSE]
+  slot(x, "fileNames", check = FALSE) <- x@fileNames[unique(x@frames$file)]
+  x
+}) 
+
+# Not sure about this but maybe this needed because the setValidity of MsBackend 
```

Comment:
that's OK. It should return a `character` with length equal to the total number of spectra. So it's implementation is OK.